src/main/haskell/kore/src/Kore/Step/Substitution.hs

```diff
@@ -92,22 +70,68 @@ mergeSubstitutions tools first second = do
     -- TODO(virgil): Return the actual condition here.
     return (makeTruePredicate, substitution, proof)
 
+-- | Merge and normalize two unification substitutions
+mergeAndNormalizeSubstitutions
+    ::  ( MetaOrObject level
+        , Ord (variable level)
+        , OrdMetaOrObject variable
+        , SortedVariable variable
+        , Show (variable level)
+        , IntVariable variable
+        , Hashable variable
+        )
+    => MetadataTools level StepperAttributes
+    -> UnificationSubstitution level variable
+    -> UnificationSubstitution level variable
+    -> Either
+         ( UnificationOrSubstitutionError     level variable )
+         ( IntCounter ( PredicateSubstitution level variable
+                      , UnificationProof      level variable ))
+mergeAndNormalizeSubstitutions tools first second = do
+    (predSubstitution, proof) <- do -- IntCounter Monad
+      { (substitutionList, proof) <-
+          normalizeSubstitutionDuplication' (first ++ second)
+      ; predSubstitution <- normalizeSubstitution' substitutionList
+      ; return (predSubstitution, proof)
+          -- TODO(virgil): Return the actual condition here. and proofs
+      }
+    return $ do --IntCounter Monad
+      { PredicateSubstitution
+        { predicate = _condition
+        , substitution = substitutionList'
+        } <- predSubstitution
+      -- ; condition' <- fst $ mergeConditionsWithAnd --TODO: merge conditions
+      --                 [ makeTruePredicate
+      --                 , condition ]
+      ; return ( PredicateSubstitution
+                 { predicate = makeTruePredicate
```

Comment:
You should not be dropping the _condition, the merged substitution is not valid without it. I.e. if you merge `[x=f(a)]` with `[x=g(b)]` you should get `[x=f(a)]` and `f(a)=g(b)` (this particular example is not implemented yet, but there's a task for it).

Reply:
Oh you are right, I fixed that, but the test still doesn't pass.  I think I'm not catching an error in `BaseStep` that should be caught (and converted to a `Bottom`)